PL2/PL2-parser/src/main/java/nl/tudelft/pl2016gr2/parser/controller/FullGfaReader.java

```diff
@@ -3,153 +3,217 @@
+        switch (chars[0]) {
+          case 'L':
+            parseEdge(chars);
+            break;
+          case 'S':
+            parseNode(chars);
+            break;
+          case 'H':
+            parseHeader(chars);
+            break;
+          default:
+        }
+      }
+    } catch (IOException ex) {
+      Logger.getLogger(FullGfaReader.class.getName()).log(Level.SEVERE, null, ex);
     }
   }
-  
+
   /**
-   * Read the actual data.
+   * Parse a line of the GFA file which contains an edge.
+   *
+   * @param chars characters of the line.
    */
-  private void read() {
-    int count = 0;
-    while (sc.hasNext()) {
-      count++;
-      String next = sc.next();
-      switch (next) {
-        case "H":
-          handleH(sc);
-          break;
-        case "S":
-          handleS(sc);
-          break;
-        case "L":
-          handleL(sc);
-          break;
-        default:
-          System.out.println("Not S, H or L at: " + count);
-          sc.nextLine();
-      }
+  private void parseEdge(char[] chars) {
+    int from = 0;
+    int index = 2;
+    while (chars[index] != '\t') {
+      from *= 10;
+      from += chars[index++] - 48;
     }
-    System.out.println("Number of lines read: " + count);
+    index += 3;
+    int to = 0;
+    while (chars[index] != '\t') {
+      to *= 10;
+      to += chars[index++] - 48;
+    }
+    getNode(to).addInlink(from);
+    getNode(from).addOutlink(to);
   }
 
   /**
-   * Private method for handling an "H" as first character on the line of a gfa file.
-   * @param sc The scanner which is used to parse the file.
+   * Parse a line of the GFA file which contains a node.
+   *
+   * @param chars characters of the line.
    */
-  private void handleH(Scanner sc) {
-    String line = sc.nextLine();
-    this.genoms = extractGenomes(line);
+  private void parseNode(char[] chars) {
+    int nodeId = 0;
+    int index = 2;
+    while (chars[index] != '\t') {
+      nodeId *= 10;
+      nodeId += chars[index++] - 48;
+    }
+    index++;
+
+    Node node = getNode(nodeId);
+
+    index = parseNodeBases(node, chars, index);
+    index = parseNodeGenoms(node, chars, index);
+    parseNodeOrientation(node, chars, index);
+  }
 
+  /**
+   * Parse the bases of a node.
+   *
+   * @param node     the node to add the bases to.
+   * @param chars    the character array of the node line in the GFA file.
+   * @param curIndex the current index; where the bases start in the line.
+   * @return the new index, after reading the genoms.
+   */
+  private int parseNodeBases(Node node, char[] chars, int curIndex) {
+    int index = curIndex;
+    index = skipTillCharacter(chars, index, '\t', 1);
+    node.setBases(new String(chars, curIndex, index - curIndex));
+    return index;
   }
 
   /**
-   * Private method for handling an "S" as first character on the line of a gfa file.
-   * @param sc The scanner which is used to parse the file.
+   * Parse the genoms of a node.
```

Comment:
Please Ctrl+R on
`genoms -> genomes`

Let's stop using 'genoms' everywhere in the code. I cry everytime I have to use `getGenoms()`.
Singular: genome
Plural: genomes

https://en.wiktionary.org/wiki/genome#English